src/components/UserInfo/UserInfo.tsx

```diff
@@ -1,4 +1,15 @@
+import React from 'react';
+import { User } from '../../types/User';
+
+interface Props {
+  user: User | null;
```

Comment:
user is always present, 'cause you do not render a component if it's not, so type this props as User
```suggestion
  user: User;
```